### PR TITLE
Project documentation and submission

### PR DESCRIPTION
**Pull request by ksigei**
### In this milestone, I was able:

- Document my project
- Ensure all functionalities are implemented
- Ensure there are no conflicts
- Ensure all tests are running
- Deploy to Heroku 
- Record a loom presentation
- [Heroku](https://budget5050.herokuapp.com/)
- [Loom Link](https://www.loom.com/share/6e2ffa040537423fa3de5bc335010b47)

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (medaminedev66) [inline app/views/devise/registrations/new.html.erb]: - [ ] Following the project requirements:
* The user should be able to register in the app with full name, email and password (all mandatory). Kindly add a field for the full name to the registration form.
* When the user logs in, they should be taken to the categories page.
- [ ] In the splash screen, the app should have a simple page with the name of your app with links to the sign up and log in pages.

File: app/views/entities/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  <% end %>
+  <div class="flex justify-between items-center">
+    <h2 class="font-bold text-4xl">Transactions for <span class="text-blue-600"><%= @group.name %></span> Category</h2>
+    </div>
+
+  <% if @entities.count > 0 %>
+  <div id="entities" class="min-w-full">
+    <%= render @entities %>
+  </div>
+
```
Comment (medaminedev66) [inline app/views/entities/index.html.erb:15]: - [ ] Following the project requirements, the list of transactions should be ordered by the most recent.

File: app/views/entities/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<div class="mx-auto md:w-2/3 w-full">
+  <h1 class="font-bold text-4xl">New entity</h1>
+
+  <%= render "form", entity: @entity %>
+
+  <%= link_to 'Back to entities', entities_path, class: "ml-2 rounded-lg py-3 px-5 bg-gray-100 inline-block font-medium" %>
```
Comment (medaminedev66) [inline app/views/entities/new.html.erb:6]: - [ ] When I click on the back button in the transactions/new form, it shows me the error below. Please ensure when the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

image

File: app/views/entities/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  <div class="my-5">
+    <%= form.label :amount %>
+    <%= form.number_field :amount, class: "block shadow rounded-md border border-gray-200 outline-none px-3 py-2 mt-2 w-full" %>
+  </div>
+
+  <div class="my-5">
+   <%= form.label :group_id %>
+    <%= form.collection_select :group_id, current_user.groups, :id, :name, {}, class: "block shadow rounded-md border border-gray-200 outline-none px-3 py-2 mt-2 w-full" %>
+  </div>
```
Comment (medaminedev66) [inline app/views/entities/_form.html.erb:27]: - [ ] The relationship between categories and transactions is a many-to-many relationship, and that means that the transaction can belong to many categories.
As required in the project requirements, the user should be able to select more than one category to which the transaction should belong.

File: app/assets/stylesheets/application.tailwind.css [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+@tailwind components;
+@tailwind utilities;
+
+/*
+
+@layer components {
+  .btn-primary {
+    @apply py-2 px-4 bg-blue-200;
+  }
+}
```
Comment (medaminedev66) [inline app/assets/stylesheets/application.tailwind.css]: - [ ] Kindly remove the commented code to keep your code clean.
- [ ] Following the project requirements, you should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
* Colors
* Typography: font face, size and weight.
* Layout: composition and space between elements.
- [ ] Please ensure that your web pages matches the design below:
* Registration page:
image

* Categories page:
image

* Transactions page: 
image

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @ksigei  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- The repo looks professional.
- You have a descriptive PR.
- Your web app is deployed properly.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**